usr/lib/api/api_interface.c

```diff
@@ -37,6 +37,28 @@
 
 void api_init();
 
+#if OPENSSL_VERSION_PREREQ(3, 0)
+#define BEGIN_OPENSSL_LIBCTX(ossl_ctx, rc)                                  \
+        do {                                                                \
+            OSSL_LIB_CTX  *prev_ctx = OSSL_LIB_CTX_set0_default((ossl_ctx));\
+            if (prev_ctx == NULL) {                                         \
+                (rc) = CKR_FUNCTION_FAILED;                                 \
+                TRACE_ERROR("OSSL_LIB_CTX_set0_default failed\n");          \
+                break;                                                      \
+            }
+
+#define END_OPENSSL_LIBCTX(rc)                                              \
+            if (OSSL_LIB_CTX_set0_default(prev_ctx) == NULL) {              \
+                if ((rc) == CKR_OK)                                         \
+                    (rc) = CKR_FUNCTION_FAILED;                             \
+                TRACE_ERROR("OSSL_LIB_CTX_set0_default failed\n");          \
+            }                                                               \
+        } while (0);
+#else
+#define BEGIN_OPENSSL_LIBCTX(ossl_ctx, rc)
+#define END_OPENSSL_LIBCTX(rc)
```

Comment:
To enforce consistent usage between OpenSSL 1.1.1 and 3.0 I would define these two macros also as start and end of a scope (either `{` and `}` or `do {` and `} while (0);`.  Makes it easier to spot compile errors later one if one person changes code with 1.1.1 and that code does not compile due to wrong scoping on 3.0.

Reply:
good point, will change.